### PR TITLE
Adds zstorbench

### PR DESCRIPTION
**Pull request by chrisvdg**
Adds the benchmark client: zstorbench

### CODE REVIEW

**Comment by GlenDC** [comment]
Will review this once PR #465 is finished and merged into master. And until all the (required) changes in there are also applied/ported over to here.

File: cmd/zstorbench/README.md [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+      --out-benchmark string       path and filename where benchmarking results are written (default "benchmark.yaml")
+      --out-profile string         path where profiling files are written (default "./profile")
+      --profile-mode profileMode   Enable profiling mode, one of [cpu, mem, block, trace]
+```
+
+## Benchmark config
+
+Client config contains a list of scenarios. 
+Each scenario is associated with a corresponding scenarioID and contains 2 main fields: 
+* `zstor_config` representing the config for the zstor client
```
[the remark this answers is not in the document]
Comment (chrisvdg) [reply cmd/zstorbench/README.md]: This is already fixed to both have the `_config` suffix, just wasn't updated in the readme text itself

[the remark this answers is not in the document]
Comment (GlenDC) [reply cmd/zstorbench/README.md]: Fair enough. Either way, I would drop the `_config` suffix, as it's all part of a config file, hence it's all config anyhow. And rename `bench` to `benchmark` as that's more clear imho.

File: benchmark/bencher/client.go [hunk 2]
```diff
@@ -0,0 +1,193 @@
+/*
```
[the remark this answers is not in the document]
Comment (GlenDC) [reply benchmark/bencher/client.go]: This comment is now tracked as an issue by @chrisvdg as #496

File: benchmark/bencher/client.go [hunk 3]
```diff
@@ -142,3 +153,41 @@ func createMetastorClientFromConfigAndDatabase(cfg *client.MetaStorConfig, db me
...
+	}
+	tlsConfig := &tls.Config{
+		MinVersion: config.MinVersion.VersionTLSOrDefault(tls.VersionTLS11),
+		MaxVersion: config.MaxVersion.VersionTLSOrDefault(tls.VersionTLS12),
+	}
+
+	if config.ServerName != "" {
+		tlsConfig.ServerName = config.ServerName
+	} else {
+		log.Warning("TLS is configured to skip verificaitons of certs, " +
```
Comment (GlenDC) [inline benchmark/bencher/client.go]: I made a typo here apparently, which you copy-pasted @chrisvdg. Anyhow, not important for now, we can fix it as part of #496.

**Pull request merged**